### PR TITLE
Return TLS cipher suites on OS older than Windows Server 2016

### LINKED ISSUE #1141: [Work Item] HealthChecker - Make the TLS Cipher Suites check available on Windows < 2016
**Describe the work**
We should check if it's possible to query the TLS Cipher Suites information from the registry (on machines where the `Get-TLSCipherSuite` cmdlet isn't available).

Registry key: `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Cryptography\Configuration\Local\SSL\00010002`
Value: `Functions`

#849

### PR DESCRIPTION
**Pull request by lusassl-msft**
**Issue:**
`Get-TlsCipherSuite` isn't available on OS older than Windows Server 2016 and so, no TLS cipher suites were displayed. This PR adds an improvement to query the cipher suites in these cases directly via registry. It also handles cases in which the TLS cipher suite order was configured via GPO and so takes precedence over the local configuration.

**Fix:**
Resolve #1141 
Query TLS cipher suites on legacy OS via registry call.

**Validation:**
Lab / Unit test

### CODE REVIEW

File: Shared/TLS/Get-TlsCipherSuiteInformation.ps1 [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    begin {
+        Write-Verbose "Calling: $($MyInvocation.MyCommand)"
+    }
+    process {
+        try {
+            # 'Get-TlsCipherSuite' takes account of the cipher suites which are configured by the help of GPO.
+            # No need to query the ciphers defined via GPO if this call is successful.
+            Write-Verbose "Trying to query TlsCipherSuites via 'Get-TlsCipherSuite'"
+            $tlsCipherSuites = Get-TlsCipherSuite -ErrorAction Stop
+        } catch {
```
Comment (dpaulson45) [inline Shared/TLS/Get-TlsCipherSuiteInformation.ps1:21]: This isn't getting the information remotely. This needs to be addressed.

File: Shared/TLS/Get-TlsCipherSuiteInformation.ps1 [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                    ValueType           = "MultiString"
+                    CatchActionFunction = $CatchActionFunction
+                }
+
+                $tlsCipherSuites = Get-RemoteRegistryValue @tlsRegistryParams
+            }
+        }
+    }
+    end {
+        return $tlsCipherSuites
```
Comment (dpaulson45) [inline Shared/TLS/Get-TlsCipherSuiteInformation.ps1:60]: Should maybe think about returning a standard object that works for both `Get-TlsCipherSuite` and the values from the registry.

File: Shared/TLS/Get-TlsCipherSuiteInformation.ps1 [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            $tlsCipherSuites = Get-TlsCipherSuite -ErrorAction Stop
+        } catch {
+            # If we can't get the ciphers via cmdlet, we need to query them via registry call and need to check
+            # if ciphers suites are defined via GPO as well. If there are some, these take precedence over what
+            # is in the default location.
+            Write-Verbose "Failed to query TlsCipherSuites via 'Get-TlsCipherSuite' fallback to registry"
+
+            if ($null -ne $CatchActionFunction) {
+                & $CatchActionFunction
+            }
```
Comment (dpaulson45) [inline Shared/TLS/Get-TlsCipherSuiteInformation.ps1:29]: Should use `Invoke-CatchActionError` from share

**Review by dpaulson45 — CHANGES_REQUESTED** [review]

File: Shared/TLS/Get-TlsCipherSuiteInformation.ps1 (hunk 1 above)
Comment (lusassl-msft) [reply Shared/TLS/Get-TlsCipherSuiteInformation.ps1:21]: Oops, you're right. Good catch.

File: Shared/TLS/Get-TlsCipherSuiteInformation.ps1 (hunk 3 above)
Comment (lusassl-msft) [reply Shared/TLS/Get-TlsCipherSuiteInformation.ps1:29]: Removed this part as I changed the logic.

File: Shared/TLS/Get-TlsCipherSuiteInformation.ps1 (hunk 2 above)
Comment (lusassl-msft) [reply Shared/TLS/Get-TlsCipherSuiteInformation.ps1:60]: We now return a standard object with the information that we need. Additional properties can be added on a need to use base.

File: Shared/TLS/Get-TlsCipherSuiteInformation.ps1 (hunk 1 above)
Comment (lusassl-msft) [reply Shared/TLS/Get-TlsCipherSuiteInformation.ps1:21]: This call is now performed via `Invoke-ScriptBlockHandler`

**Review by dpaulson45 — APPROVED** [review]

**Pull request merged**